service/controller/clusterapi/v29/resource/clusterazs/create.go

```diff
@@ -48,30 +65,155 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d MachineDeployments for tenant cluster", len(machineDeployments)))
 	}
 
-	var azs []string
+	var azs map[string]subnetPair
 	{
-		var azsMap map[string]struct{}
+		// Acquire AZs together with corresponding subnets from AWS EC2 API
+		// results.
+		azs, err = fromEC2SubnetsToMap(cc.Status.TenantCluster.TCCP.Subnets)
```

Comment:
> Are we sure cc.Status.TenantCluster.TCCP.Subnets already holds the right information? 

We are sure that before CF stack creation it doesn't hold anything. At least that's the expectation.

> The code below to ensure the master subnet in there is a bit odd already. It would be nice to have once place to hold the subnets.

This code ensures that AZs configured for specific tenant cluster are assigned with correct subnets. When the AZ and corresponding subnet already exists in AWS, the connection must be kept. For rest of the AZs (the new ones that don't have resources in AWS yet) new subnet must be assigned from available ones.

> When this resource here is this place, I would question why we even need to use anything from cc.Status.TenantCluster.TCCP.Subnets.

Slice of AWS state of TCCP subnets is crucial for functionality of this code. This code wouldn't exist without and dynamic AZs wouldn't be possible without. Therefore it's needed.